### PR TITLE
Searching Fix Up

### PR DESCRIPTION
**Pull request by Unevilicorn**
Instance method for searching functions
Test for searching functions
Workflow for test_searching

### CODE REVIEW

File: ivy/functional/backends/jax/searching.py [hunk 1]
```diff
@@ -30,11 +31,9 @@ def argmin(
...
 
-def nonzero(x: JaxArray, *, out: Optional[JaxArray] = None) -> JaxArray:
+def nonzero(x: JaxArray) -> Tuple[JaxArray]:
     return jnp.nonzero(x)
 
 
-def where(
-    condition: JaxArray, x1: JaxArray, x2: JaxArray, *, out: Optional[JaxArray] = None
-) -> JaxArray:
+def where(condition: JaxArray, x1: JaxArray, x2: JaxArray) -> JaxArray:
```
Comment (jiahanxie353) [inline ivy/functional/backends/jax/searching.py:38]: Please do not delete `out` as explained :)

File: ivy/functional/backends/numpy/searching.py [hunk 2]
```diff
@@ -33,11 +37,8 @@ def nonzero(x: np.ndarray) -> Tuple[np.ndarray]:
...
 
 
-def where(
-    condition: np.ndarray,
-    x1: np.ndarray,
-    x2: np.ndarray,
-    *,
-    out: Optional[np.ndarray] = None,
-) -> np.ndarray:
+def where(condition: np.ndarray, x1: np.ndarray, x2: np.ndarray) -> np.ndarray:
```
Comment (jiahanxie353) [inline ivy/functional/backends/numpy/searching.py:40]: Please do not delete `out` as explained :)

File: ivy/functional/backends/tensorflow/searching.py [hunk 3]
```diff
@@ -1,14 +1,13 @@
...
 import tensorflow as tf
-from typing import Optional, Union
 
 
 def argmax(
     x: Union[tf.Tensor, tf.Variable],
     axis: Optional[int] = None,
     keepdims: bool = False,
-    *,
-    out: Optional[Union[tf.Tensor, tf.Variable]] = None
```
Comment (jiahanxie353) [inline ivy/functional/backends/tensorflow/searching.py:11]: Please do not delete `out` as explained :)

File: ivy/functional/backends/tensorflow/searching.py [hunk 4]
```diff
@@ -20,8 +19,6 @@ def argmin(
...
```
Comment (jiahanxie353) [inline ivy/functional/backends/tensorflow/searching.py:24]: Please do not delete `out` as explained :)

File: ivy/functional/backends/tensorflow/searching.py [hunk 5]
```diff
@@ -30,15 +27,13 @@ def argmin(
...
+) -> Tuple[Union[tf.Tensor, tf.Variable]]:
+    return tuple(tf.experimental.numpy.nonzero(x))
 
 
 def where(
     condition: Union[tf.Tensor, tf.Variable],
     x1: Union[tf.Tensor, tf.Variable],
     x2: Union[tf.Tensor, tf.Variable],
-    *,
-    out: Optional[Union[tf.Tensor, tf.Variable]] = None
```
Comment (jiahanxie353) [inline ivy/functional/backends/tensorflow/searching.py:42]: Please do not delete `out` as explained :)

File: ivy/functional/backends/torch/searching.py [hunk 6]
```diff
@@ -38,13 +37,7 @@ def nonzero(x: torch.Tensor) -> Tuple[torch.Tensor]:
     return torch.nonzero(x, as_tuple=True)
 
 
-def where(
-    condition: torch.Tensor,
-    x1: torch.Tensor,
-    x2: torch.Tensor,
-    *,
-    out: Optional[torch.Tensor] = None
```
Comment (jiahanxie353) [inline ivy/functional/backends/torch/searching.py:46]: Please do not delete `out` as explained :)

**Review by jiahanxie353 — CHANGES_REQUESTED** [review]
Please see the comments :)

File: ivy/functional/backends/jax/searching.py [hunk 7]
```diff
@@ -30,7 +31,7 @@ def argmin(
 argmin.support_native_out = True
 
 
-def nonzero(x: JaxArray, *, out: Optional[JaxArray] = None) -> JaxArray:
+def nonzero(x: JaxArray, *, out: Optional[JaxArray] = None) -> Tuple[JaxArray]:
```
Comment (djl11) [inline ivy/functional/backends/jax/searching.py:34]: we should not have the `out` argument for this function. The `out` argument is only relevant for functions which return a single array.

**Review by djl11 — CHANGES_REQUESTED** [review]

**Review by djl11 — APPROVED** [review]
lgtm!

**Pull request merged**